internal/controller/application_controller.go

```diff
@@ -64,28 +67,39 @@ type ApplicationReconciler struct {
 func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = log.FromContext(ctx)
 
-	obj := &apisv1alpha1.Application{}
-	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
+	app := &apisv1alpha1.Application{}
+	if err := r.Client.Get(ctx, req.NamespacedName, app); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	var secret corev1.Secret
 	err := r.Client.Get(ctx, client.ObjectKey{
 		Namespace: req.Namespace,
-		Name:      obj.Spec.DatabaseSecretRef.Name,
+		Name:      app.Spec.DatabaseSecretRef.Name,
 	}, &secret)
 	if err != nil {
 		return ctrl.Result{
 			Requeue: true,
 		}, err
 	}
 
-	namespace, ok := obj.Annotations["kcp.io/cluster"]
+	namespace, ok := app.Annotations["kcp.io/cluster"]
 	if !ok {
 		return ctrl.Result{}, fmt.Errorf("cluster label not found")
 	}
 
-	deployment, err := getApplicationDeployment(obj, namespace)
+	var db cnpgapiv1.Database
+	err = r.ProviderClient.Get(ctx, types.NamespacedName{
```

Comment:
We need the Database object to retrieve the pg Cluster name which is needed to generate the host name (part of service name).

An alternative would be to store the pg cluster name inside the Application CRD, in which case we would no longer need cnpg client and its dependencies.

Reply:
I think its fine to have it either way. as long as it works :)